sequencers/modeshape-sequencer-epub/src/main/java/org/modeshape/sequencer/epub/EpubMetadata.java

```diff
@@ -0,0 +1,225 @@
+            ZipEntry entry = null;
+            while ((entry = zip.getNextEntry()) != null) {
+                if (entry.getName().endsWith("/content.opf")) {
+                    // get content file
+                    ByteArrayOutputStream content = new ByteArrayOutputStream();
+                    byte[] bytes = new byte[(int) entry.getSize()];
+                    int read;
+                    while ((read = zip.read(bytes, 0, bytes.length)) != -1) {
+                        content.write(bytes, 0, read);
+                      }
+                    content.flush();
+
+                    // get metadata elements
+                    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                    factory.setNamespaceAware(true);
+                    DocumentBuilder builder = factory.newDocumentBuilder();
+                    Document doc = builder.parse(new ByteArrayInputStream(content.toByteArray()));
+
+                    XPathFactory xPathfactory = XPathFactory.newInstance();
+                    XPath xpath = xPathfactory.newXPath();
+                    xpath.setNamespaceContext(new SimpleNSC(DUBLIN_CORE_PREFIX, DUBLIN_CORE_URI));
+                    XPathExpression expr = xpath.compile("//" + DUBLIN_CORE_PREFIX + ":*");
+                    NodeList metadata = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
+
+                    for (int i = 0; i < metadata.getLength(); i++) {
+                        Node node = metadata.item(i);
+                        if (node.getLocalName().equals("title")) {
+                            title = node.getTextContent();
+                        } else if (node.getLocalName().equals("creator")) {
+                            creator = node.getTextContent();
+                        } else if (node.getLocalName().equals("contributor")) {
+                            contributor = node.getTextContent();
+                        } else if (node.getLocalName().equals("language")) {
+                            language = node.getTextContent();
+                        } else if (node.getLocalName().equals("identifier")) {
+                            identifier = node.getTextContent();
+                        } else if (node.getLocalName().equals("subject")) {
+                            subject = node.getTextContent();
+                        } else if (node.getLocalName().equals("description")) {
+                            description = node.getTextContent();
+                        } else if (node.getLocalName().equals("publisher")) {
+                            publisher = node.getTextContent();
+                        } else if (node.getLocalName().equals("date")) {
+                            date = node.getTextContent();
+                        }
+                    }
+                }
+            }
+            zip.close();
+
+            return true;
+        } catch (Exception e) {
+            // log
```

Comment:
the comment is correct, this should be logged or left to bubble up and logged in the sequencer